resources/options.js

```diff
@@ -68,31 +68,76 @@ function restore_options_availability() {
         // Default value
         loggingEnabled: true
     }, function(items) {
-        // The availability depends on the value of the logging checkbox state.
-        l('tabs').disabled = l('comments').disabled =
-            l('peer_comments').disabled = l('focus').disabled =
-                l('username').disabled = l('repo').disabled =
-                    l('file').disabled = !items.loggingEnabled;
+        var enabled = items.loggingEnabled;
+        if (enabled) {
+            show();
+        } else {
+            hide();
+        }
     });
 }
 
+// Shows the sub-options.
+function show_sub() {
+    $('#security_sub').show();
+    $('#privacy_sub').show();
+    $('#hints_sub').show();
+}
+
+// Hides the sub-options.
+function hide_sub() {
+    $('#security_sub').hide();
+    $('#privacy_sub').hide();
+    $('#hints_sub').hide();
+}
+
+// Shows the option cards.
+function show() {
+    $('#security').show();
+    $('#privacy').show();
+    $('#hints').show();
+}
+
+// Hides the option cards.
+function hide() {
+    $('#security').hide();
+    $('#privacy').hide();
+    $('#hints').hide();
+}
+
+// Switches availability of options.
+function switch_disable() {
+    l('tabs').disabled = !l('tabs').disabled;
+    l('comments').disabled = !l('comments').disabled;
+    l('peer_comments').disabled = !l('peer_comments').disabled;
+    l('focus').disabled = !l('focus').disabled;
+    l('username').disabled = !l('username').disabled;
+    l('repo').disabled = !l('repo').disabled;
+    l('file').disabled = !l('file').disabled;
+}
+
+// Restores availability of options.
+// The availability depends on the value of the logging checkbox state.
+function restore_disable() {
+    l('tabs').disabled = l('comments').disabled =
+        l('peer_comments').disabled = l('focus').disabled =
+            l('username').disabled = l('repo').disabled =
+                l('file').disabled = !items.loggingEnabled;
+}
+
+
 // Disables the (sub)options when the user doesn't want Octopeer to log data and
 // enables the (sub)options when the user does.
 // A message will confirm this action to the user.
 function switch_options() {
     var logging = l('logging').checked;
     if (logging) {
         Materialize.toast("Logging has been enabled.", 2000);
+        show();
     } else {
         Materialize.toast("Logging has been disabled.", 2000);
+        hide();
```

Comment:
I see the same show and hide methods are used here. Maybe you want to make the global show() and hide() functions call one of the optional show/hide methods, so you only have to edit this on one place instead of on two when you want to change the show/hide method.